amplipi/streams.py

```diff
@@ -870,56 +870,87 @@ def connect(self, src):
       return
 
     # Generate some of the DLNA_Args
-    self.uuid = 0
-    self.uuid = uuid_gen()
+    self._uuid = 0
+    self._uuid = uuid_gen()
     portnum = 49494 + int(src)
 
-    # Make the (per-source) config directory
-    src_config_folder = f'{utils.get_folder("config")}/srcs/{src}'
-    os.system(f'mkdir -p {src_config_folder}')
+    # Make the (per-source) config and web directories
+    self._src_config_folder = f'{utils.get_folder("config")}/srcs/{src}'
+    os.system(f'rm -r {self._src_config_folder}')
+    os.system(f'mkdir -p {self._src_config_folder}')
 
-    meta_args = [f'{utils.get_folder("streams")}/dlna_metadata.bash', f'{src_config_folder}']
+    self._src_web_folder = f'{utils.get_folder("web")}/generated/{src}'
+    os.system(f'rm -r {self._src_web_folder}')
+    os.system(f'mkdir -p {self._src_web_folder}')
+
+    # Make the fifo to be used for commands
+    os.mkfifo(f'{self._src_config_folder}/cmd') # lazily open fifo so startup is faster
+
+    # startup the metadata process and the DLNA process
     dlna_args = ['gmediarender', '--gstout-audiosink', 'alsasink',
-                 '--gstout-audiodevice', utils.real_output_device(src), '--gstout-initial-volume-db',
-                 '0.0', '-p', f'{portnum}', '-u', f'{self.uuid}',
-                 '-f', f'{self.name}', '--logfile',
-                 f'{src_config_folder}/metafifo']
-    self.proc = subprocess.Popen(args=meta_args, preexec_fn=os.setpgrp, stdin=subprocess.PIPE,
-                                 stdout=subprocess.PIPE, stderr=subprocess.PIPE)
-    self.proc2 = subprocess.Popen(args=dlna_args, stdin=subprocess.PIPE, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+                '--gstout-audiodevice', utils.real_output_device(src), '--gstout-initial-volume-db',
+                '0.0', '-p', f'{portnum}', '-u', f'{self._uuid}',
+                '-f', f'{self.name}']
+    meta_args = [ sys.executable,
+                 f'{utils.get_folder("streams")}/dlna_meta.py',
+                 f'{self.name}',
+                 f'{self._src_config_folder}/cmd',
+                 f'{self._src_config_folder}/meta.json',
+                 self._src_web_folder,
+                ]
+                # '-d']
+
+    self.proc = subprocess.Popen(args=dlna_args, stdin=subprocess.PIPE, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+    self._metadata_proc = subprocess.Popen(args=meta_args, stdin=subprocess.PIPE, stdout=sys.stdout, stderr=sys.stderr)
```

Comment:
If this process contains loggers, send the logs to the sys.sdtout (and errors to sys.stderr) directly as I do in LMS._activate() around line 1373:

`self.meta_proc = subprocess.Popen(args=meta_args, stdout=sys.stdout, stderr=sys.stderr)`

If the logs still get output to `journalctl -f` with PIPE, feel free to not change this